### DISCUSSION TITLE
v3.2.87

### DISCUSSION THREAD

**Discussion by danharrin** [filamentphp/filament#13177] [General]
## What's Changed
* change tabindex for login form by @pxlrbt in https://github.com/filamentphp/filament/pull/13111
* Ignore textarea style height by default, not just on auto size by @nielsvh2103 in https://github.com/filamentphp/filament/pull/13102
* Fix getting infolist action in test methods by @zepfietje in https://github.com/filamentphp/filament/pull/13107
* Feature: Shift click to select table records (like Gmail and Linear) by @neverything in https://github.com/filamentphp/filament/pull/13074
* Improve Malaysian translations by @alifdarsim in https://github.com/filamentphp/filament/pull/13142
* Fix text wrapping for bulleted list by @howdu in https://github.com/filamentphp/filament/pull/13041
* Prevent infolist entry grid blowout by @ralphjsmit in https://github.com/filamentphp/filament/pull/13125
* Support badge size for action components by @tnylea in https://github.com/filamentphp/filament/pull/12784
* Prevent hidden wizard steps to be focusable by @sebapastore in https://github.com/filamentphp/filament/pull/13150
* Adding assertFormComponentExists() test method for forms by @cheesegrits in https://github.com/filamentphp/filament/pull/13154
* Add missing class string type annotations by @juliangums in https://github.com/filamentphp/filament/pull/13130
* Support previews in builder by @awcodes in https://github.com/filamentphp/filament/pull/12629
* Fix wrong type in docs by @thiagolcks in https://github.com/filamentphp/filament/pull/13165
* docs: Removed extra parenthesis in modifyQueryUsing by @victorelec14 in https://github.com/filamentphp/filament/pull/13167
* Add fallback default locale for format curreny by @datlechin in https://github.com/filamentphp/filament/pull/13166
* Update Dutch builder translations by @zepfietje in https://github.com/filamentphp/filament/pull/13173

## New Contributors
* @nielsvh2103 made their first contribution in https://github.com/filamentphp/filament/pull/13102
* @neverything made their first contribution in https://github.com/filamentphp/filament/pull/13074
* @alifdarsim made their first contribution in https://github.com/filamentphp/filament/pull/13142
* @thiagolcks made their first contribution in https://github.com/filamentphp/filament/pull/13165
* @victorelec14 made their first contribution in https://github.com/filamentphp/filament/pull/13167

**Full Changelog**: https://github.com/filamentphp/filament/compare/v3.2.86...v3.2.87

<hr /><em>This discussion was created from the release <a href='https://github.com/filamentphp/filament/releases/tag/v3.2.87'>v3.2.87</a>.</em>

**Comment by ekrist1** [comment]
This release seems to break navigation using SPA mode:
Could not find Livewire component in DOM tree
Unhandled Promise Rejection: Could not find Livewire component in DOM tree

This happens when you navigate between menu items. A full refresh will load the page.

**Reply by danharrin** [reply]
@ekrist1 could you please investigate which PR causes this issue, since it's not possible to undo the entire release

**Reply by neverything** [reply]
@danharrin @ekrist1 seems like I missed testing the table feature in spa mode. Sorry about that.

**Reply by danharrin** [reply]
@neverything are you going to be able to handle that soon or do I need to revert the PR?

**Reply by neverything** [reply]
@danharrin I'm trying, but it seems to be a bit more complicated then expected. So sadly, it might be best to revert the PR for now. Super sorry, that I didn't catch it earlier.

**Reply by neverything** [reply]
@danharrin a quick fix that seems to work with both lazy loaded tables and wire:navigate is to not compare `component.id === this.$wire.id`: 

Something like this:
```javascript
Livewire.hook('element.init', () => {
    this.watchForCheckboxClicks()
})
```
instead of:
```javascript
Livewire.hook('element.init', ({ component }) => {
    if (component.id === this.$wire.id) {
        this.watchForCheckboxClicks()
    }
})
```

But doing this on every `element.init` might be a bit crazy and I haven't found an easier way yet.

**Reply by bzy107** [reply]
The same issue seems to occur when switching tabs with multiple RelationManagers.
If this problem is not resolved, it would be better to revert #13074.

https://github.com/filamentphp/filament/assets/28666304/fdb31fdc-a3e5-43cd-9259-bade4e25a465

**Reply by neverything** [reply]
Thanks for the report. Played around with it some more, but couldn't find an easier solution yet other than the one I mentioned running on every `element.init`

**Reply by neverything** [reply]
@danharrin I think I found a more promising solution, but I need to test it with tabbed relationship managers

```javascript
Livewire.hook('element.init', ({ component }) => {
    if (component.id !== this.$root.closest('[wire\\:id]').attributes['wire:id'].value) {
        return
    }

    this.$nextTick(() => this.watchForCheckboxClicks())
})
```

Somehow getting the `wire:id` of the element this way works

**Reply by danharrin** [reply]
Okay, please let me know how you get on, if we don't have a merged solution in a couple of hours I am going to revert

**Reply by neverything** [reply]
@danharrin was able to test in spa mode and relationship manager. Created https://github.com/filamentphp/filament/pull/13186

**Reply by danharrin** [reply]
Should be fixed in the latest release

**Reply by bzy107** [reply]
The tabbed RelationManagers are working correctly.
I appreciate your prompt responses.
